test/renate/utils/test_hf_utils.py

```diff
@@ -0,0 +1,59 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+import pytest
+import torch
+from transformers import AutoTokenizer
+
+from renate.utils.hf_utils import BatchEncoding, DataCollatorWithPaddingForWildTime
+
+
+@pytest.mark.parametrize(
+    "features, shape, error",
+    [
+        (
+            [
+                ({"input_ids": [0, 1, 2]}, 0),
+                ({"input_ids": [0, 1, 2, 3, 4, 5]}, 1),
+            ],
+            torch.Size([2, 6]),
+            False,
+        ),
+        (
+            [
+                (({"input_ids": [1, 2, 4]}, 0), {}),
+                (({"input_ids": [4, 5, 6, 7, 8, 9]}, 1), {}),
+            ],
+            torch.Size([2, 6]),
+            False,
+        ),
+        (
+            [
+                (({"input_ids": [1, 2, 4]}, 0), [1, 2]),
+                (({"input_ids": [4, 5, 6, 7, 8, 9]}, 1), [5, 6]),
```

Comment:
is `[5, 6]` the metadata? If yes, that's not the right format. It will be a dict.